src/main/scala/org/fortysevendeg/sparksftp/ReadingSFTPConnectorApp.scala

```diff
@@ -28,47 +28,81 @@ object ReadingSFTPConnectorApp extends IOApp {
           config.spark.serializer.contains("KryoSerializer").toString
         )
         .set("spark.hadoop.fs.sftp.impl", "org.apache.hadoop.fs.sftp.SFTPFileSystem")
+        .set("spark.sql.legacy.allowCreatingManagedTableUsingNonemptyLocation","true")
         .registerKryoClasses(RegisterInKryo.classes.toArray)
 
       sparkSession: SparkSession = SparkSession.builder
         .config(defaultSparkConf)
         .enableHiveSupport
         .getOrCreate()
 
+      sftpUser1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_USER")
+        .getOrElse(config.sftp.sftpUser)
+      sftpPass1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PASS")
+        .getOrElse(config.sftp.sftpPass)
+      sftpHost1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_HOST")
+        .getOrElse(config.sftp.sftpHost)
+      sftpPath1 = sparkSession.sparkContext.getConf
+        .getOption("spark.executorEnv.SFTP_PATH")
+        .getOrElse(config.sftp.sftpPath)
```

Comment:
For me, all this configuration reading logic could be placed in the config module that you already have

Reply:
Changed in d2734cd84776a08db06bbaa45e87bc268135861c